identity/src/lib/rpc/server.rs

```diff
@@ -227,109 +227,54 @@ impl IdentityService for IdentityServer {
         _: context::Context,
         code: AuthorizationCode,
     ) -> RpcResult<SessionToken> {
-        use crate::db::schema::users::dsl::*;
-
-        let authorization_code = match oauth::AuthorizationCode::new(code) {
-            Ok(val) => val,
-            Err(_) => return Err(Error::InvalidData),
-        };
+        // Checks if the authorization code has a valid form
+        let authorization_code = oauth::AuthorizationCode::new(code)?;
```

Comment:
Why not import the structs from `oauth` either explicitly `oauth::AuthorizationCode` or as `oauth::*`.

```suggestion
        let authorization_code = AuthorizationCode::new(code)?;
```